driver/src/orderbook/mod.rs

```diff
@@ -59,13 +60,15 @@ impl<'a> StableXOrderBookReading for PaginatedStableXOrderBookReader<'a> {
         let block = self.web3.eth().block_number().wait()?.as_u64();
         let mut reader = PaginatedAuctionDataReader::new(index);
         loop {
-            let number_of_orders = reader.apply_page(&self.contract.get_auction_data_paginated(
-                block,
-                self.page_size,
-                reader.pagination().previous_page_user,
-                reader.pagination().previous_page_user_offset as u64,
-            )?);
-            if (number_of_orders as u64) < self.page_size {
+            let number_of_orders: u16 = reader
+                .apply_page(&self.contract.get_auction_data_paginated(
+                    block,
+                    self.page_size,
+                    reader.pagination().previous_page_user,
+                    reader.pagination().previous_page_user_offset.try_into()?,
+                )?)
+                .try_into()?;
```

Comment:
nit: should we expect here since apply_page should never return more requested?

Reply:
Good point!